### PR TITLE
Add user token support

### PR DESCRIPTION
**Pull request by emirsavran**

### CODE REVIEW

File: src/lib/HypeDayReact.tsx [hunk 1]
```diff
@@ -5,20 +5,23 @@ import OpenResponseSection from "./components/OpenResponseSection";
...
 import WalletSection from "./components/WalletSection";
-import { getProject } from "./api";
+import { getProject, setAuthorizationHeader } from "./api";
 import "./style.css";
 import RegisterButton from "./components/RegisterButton";
 
 interface HypeDayReactProps {
   appId: string;
   projectId: string;
   walletAddress?: string;
```
Comment (kurtprp) [inline src/lib/HypeDayReact.tsx]: I agree lets remove wallet address.

**Comment by kurtprp** [comment]
@emirsavran do we need to update the readme file?

File: src/lib/HypeDayReact.tsx [hunk 2]
```diff
@@ -78,17 +85,20 @@ export default function HypeDayReact({
         projectData={projectData}
         appId={appId}
         walletAddress={walletAddress}
+        doesTokenExist={!!userToken}
```
Comment (kurtprp) [inline src/lib/HypeDayReact.tsx]: I usually shorten these to tokenExist, but this is fine as well.

File: src/lib/api.ts [hunk 3]
```diff
@@ -1,6 +1,15 @@
 import type { SupportedChains } from "./types";
 
 const BASE_URL = "https://apitest.hype.day";
+const headers = new Headers({
+  "Content-Type": "application/json",
+});
+
+export function setAuthorizationHeader(token: string | undefined) {
+  token
+    ? headers.set("Authorization", `Bearer ${token}`)
```
Comment (kurtprp) [inline src/lib/api.ts:8]: will this delete the content-type?

**Review by kurtprp — APPROVED** [review]
I agree with removing the wallet to simplify

File: src/lib/api.ts (hunk 3 above)
Comment (emirsavran) [reply src/lib/api.ts:8]: No, it only adds / removes `authorization` header

File: src/lib/HypeDayReact.tsx (hunk 2 above)
Comment (emirsavran) [reply src/lib/HypeDayReact.tsx]: I saw that I use `doesTokenExist` to identify if we have any users or not. So I changed it to `hasUser`

**Review by kurtprp — APPROVED** [review]

**Comment by kurtprp** [comment]
LGTM

**Pull request merged**